tests/test-wibox-shape.lua

```diff
@@ -18,14 +18,103 @@ local wb = wibox {
 wb:geometry(screen[1].geometry)
 wb.visible = true
 
-runner.run_steps({
+local count = 0
+
+local steps = {
     function()
         assert(wb.shape == shape.powerline)
         assert(wb.shape_bounding) -- This is a memory leak! Don't copy!
         if was_drawn then
             return true
         end
+    end,
+    -- Remove the shape and test the input
+    function()
+        wb.shape = nil
+        wb.input_passthrough = false
+        wb:connect_signal("button::press", function()
+            count = count + 1
+        end)
+
+        wb:geometry {
+            x      = 0,
+            y      = 100,
+            width  = 101,
+            height = 101,
+        }
+        wb.border_width = 0
+
+        return true
     end
-})
+}
+
+local function click(x, y)
+    table.insert(steps, function()
+        mouse.coords{x=x, y=y}
+        return true
+    end)
+    table.insert(steps, function()
+        assert(mouse.coords().x == x and mouse.coords().y == y)
+        root.fake_input("button_release", 1)
+        root.fake_input("button_press", 1)
+        return true
+    end)
+end
+
+local function check_count(cnt)
+    table.insert(steps, function()
+        if cnt == count then
+            return true
+        end
+    end)
+end
+
+-- Check each corner
+click(0 , 100)
+click(99, 100)
+click(99, 199)
+click(0 , 199)
+
+-- Batch the count check because they fail a couple time each
+check_count(4)
+
+table.insert(steps, function()
+    wb.input_passthrough = true
+    count = 0
+    awesome.sync()
```

Comment:
This sync should not be necessary. I assume it is for making sure that `input_passthrough` was handled by the X server, but that should not be necessary. I think you just need a `sync()` when you wait for events generated by the X server.